lib/factory_bot/attribute_assigner.rb

```diff
@@ -37,8 +37,9 @@ def method_tracking_evaluator
     end
 
     def decorated_evaluator
-      Decorator::InvocationTracker.new(
-        Decorator::NewConstructor.new(@evaluator, @build_class)
+      Decorator::NewConstructor.new(
```

Comment:
Ha, clever! Yeah, I'm good with this change.

I want to say there might still be an edge case if you have `add_attribute(:new)` and then try to use the `new` attribute inside `initialize_with`, but I think that's an acceptable edge case.